roles/kubernetes/preinstall/tasks/0090-etchosts.yml

```diff
@@ -5,7 +5,7 @@
       {% for item in (groups['k8s_cluster'] + groups['etcd']|default([]) + groups['calico_rr']|default([]))|unique -%}
       {% if 'access_ip' in hostvars[item] or 'ip' in hostvars[item] or 'ansible_default_ipv4' in hostvars[item] -%}
       {{ hostvars[item]['access_ip'] | default(hostvars[item]['ip'] | default(hostvars[item]['ansible_default_ipv4']['address'])) }}
-      {%- if ('ansible_hostname' in hostvars[item] and item != hostvars[item]['ansible_hostname']) %} {{ hostvars[item]['ansible_hostname'] }}.{{ dns_domain }} {{ hostvars[item]['ansible_hostname'] }}{% endif %} {{ item }}.{{ dns_domain }} {{ item }}
+      {%- if ('ansible_hostname' in hostvars[item] and item != hostvars[item]['ansible_hostname']) %} {{ hostvars[item]['ansible_hostname'] }}.{{ dns_domain }} {{ hostvars[item]['ansible_hostname'] }}{% endif %} {% if item in groups['kube_control_plane'] %}kubernetes.default.svc.{{ dns_domain }} {% endif %}{{ item }}.{{ dns_domain }} {{ item }}
```

Comment:
Will host records always be added?
this is very bad

Reply:
I can filter that case when we got a consensus about when that must works.

As an example, we can just use that as a fallback and drop {{ first_master }} entirely.

Moreover, we will have better balancing & behaviour if shuffle that DNS RR for every non-master host

Reply:
please filter this case
add to hosts only when loadbalancer_apiserver_type == 'dns-hosts'